snowshu/core/configuration_parser.py

```diff
@@ -94,8 +102,49 @@ def _get_dict_from_anything(self, dict_like_object:Union[str,'StringIO',dict])->
             try:
                 return yaml.safe_load(dict_like_object.read())
             except AttributeError:
-                with open(dict_like_object,'r') as f:
-                    return yaml.safe_load(f.read())
+                return self._load_and_verify_file(dict_like_object, **kwargs)
+
+    def _load_and_verify_file(self,
+                              file_path,
+                              verify: bool = True,
+                              templates_path: Path = Path(os.path.dirname(__file__)).parent / 'templates',
+                              schema_path: Path = None):
+        logger.debug("Parsing file at %s", file_path)
+        if not schema_path:
+            schema_path = templates_path / f'{Path(file_path).stem}_schema.yml'
+        if not os.path.isfile(schema_path):
+            logger.warning("Loading from %s, but no schema (%s) found to verify it.",
+                            file_path, schema_path)
+            verify = False
+
+        # Load data
+        try:
+            with open(file_path, 'r') as f:
+                loader = AnsibleLoader(f)
+                instance = loader.get_single_data()
+        except FileNotFoundError:
+            loader = None
+            raise
+        except Exception as exc:
+            raise Exception(f'Error when loading YAML config at {file_path} {exc}')
```

Comment:
I'm assuming the loader throws a pretty specific set of exceptions - like KeyErrors or similar right? we should tighten this down a bit if we can.